test/index.js

```diff
@@ -1,22 +1,32 @@
+/**
+ * @typedef {import('postcss').Declaration} Declaration
+ * @typedef {import('../index.js').SyntaxRule} SyntaxRule
+ */
+
 import function_ from '../index.js';
 import { runCodeTest } from './util/index.js';
 
 // @ts-ignore
-const { ruleName, messages } = function_;
+const { ruleName } = function_;
 
 runCodeTest({
 	ruleName: ruleName,
 	config: [
-		[
+		/** @type Array<SyntaxRule> */ ([
 			{
 				selector: 'rule[selector="a"]',
 				message: 'Anchors not allowed.'
 			},
 			{
 				selector: 'decl[prop="z-index"]',
 				message: 'z-index not allowed.'
+			},
+			{
+				selector: 'decl[prop="zoom"]',
+				message: (node) =>
+					`${/** @type Declaration */ (node).prop} not allowed.`
```

Comment:
@robwierzbowski we need to explictly tell it which type node is since querying will return any PostCSS node. I guess this could be adjusted in postcss-query-ast to return proper types based on selector which is passed as argument, but I’m no TS wizard for that 😄